src/conditionals/simple_programs_with_decision_points/hard/q1/MainTest.java

```diff
@@ -0,0 +1,46 @@
+package conditionals.simple_programs_with_decision_points.hard.q1;
+
+import global.BaseTest;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest extends BaseTest {
+    // Parsons with distractors
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "5";
+        return new Clause[]{
+                new StringLiteral("Enter a number: "),
+                new NewLine(),
+                new IntegerLiteral("absoluteValue")
+        };
+    }
+
+    public void runMain() {
+        AbsolutelySure.main(new String[]{});
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of(5,5), Arguments.of(-46, 46), Arguments.of(0, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(int input, int output) {
+        String error = "The absolute value was calculated incorrectly. ";
+        if(input<0 )
+            error += "Negative values were no handled properly.";
```

Comment:
I presume this should say not instead of no

Reply:
I also feel that I don't think the message is fantastic if you deliver both, maybe you should have different messages depending on what happens

Reply:
what do you mean? If it fails a test for a positive value it just says the base error message but then it adds the customized negative one if it doesn't convert the negative to a positive.